Universities

BlessingAmoah opened the pull request:
I have added a new feature to the profile update where the user has the option to select their university from list of universities in the USA fetched from this  [Universities API](https://github.com/Hipo/university-domains-list?tab=readme-ov-file). 
<img width="861" alt="Screenshot 2024-07-17 at 11 37 48 AM" src="https://github.com/user-attachments/assets/4feee7ee-a696-4a39-bb97-1b2c581e7908">
<img width="861" alt="Screenshot 2024-07-17 at 11 38 03 AM" src="https://github.com/user-attachments/assets/c640b557-6eb1-4564-b534-a8ccb34535f6">
<img width="861" alt="Screenshot 2024-07-17 at 11 38 28 AM" src="https://github.com/user-attachments/assets/e0e167e3-4873-41ea-b935-6ee94e4eb6fd">
<img width="861" alt="Screenshot 2024-07-17 at 11 38 42 AM" src="https://github.com/user-attachments/assets/b268c6e1-a229-4ff2-a51b-f204080cc048">

tpv0613 commented on Frontend/student-hub/src/pages/Profile.js:
```diff
@@ -47,10 +46,31 @@ function Profile() {
+          const data = await response.json();
+          setUniversities(data)
+        }else{
+          setError(`Failed to fetch universities: ${error.message}`);
```
`error` here could be empty or unrelated from a previous error since this isn't catching an error - let's pass in the error message from the response

tpv0613 commented on Frontend/student-hub/src/pages/Profile.js:
```diff
@@ -319,15 +340,24 @@ function Profile() {
+            <Select
               name="school"
-              value={profile.school || ''}
+              value={profile.school}
```
Can `school` be null? If so, what would the UI look like with a null school? Can users remove the school if they want to go back to not having a school?

tpv0613 reviewed (commented):
Looks good overall! Left a comment on a possible edge case for making schools be selectable. Thanks for keeping the PR succint :)

BlessingAmoah commented:
I have added an option for the user to remove their school and also added a logical or operator so that in the case  that school  option is null. 
<img width="856" alt="Screenshot 2024-07-18 at 1 58 49 PM" src="https://github.com/user-attachments/assets/6c38c72f-e0a5-4637-b4a5-df055539cb6d">
<img width="856" alt="Screenshot 2024-07-18 at 1 59 29 PM" src="https://github.com/user-attachments/assets/98cbcf0b-feea-452b-ac27-ace3df1e7574">
<img width="856" alt="Screenshot 2024-07-18 at 1 59 54 PM" src="https://github.com/user-attachments/assets/1c66519b-379f-4d1c-8b01-9ca96f61a5d6">

brittanylau commented on Frontend/student-hub/src/pages/Profile.js:
```diff
@@ -48,9 +47,31 @@ function Profile() {
+          const data = await response.json();
+          setUniversities(data)
+        }else{
+          setError(`Failed to fetch universities: ${error.message}`);
```
you can probably `throw` the error message instead, so that you catch the error in your catch block!

brittanylau reviewed (commented):
this looks great! just a small note to reduce repeated code :)
